astropy/modeling/parameters.py

```diff
@@ -178,6 +178,8 @@ class Parameter:
     bounds : tuple
```

Comment:
I recently ran into this issue of using magnitude units in model parameters too, and glad to see you guys are already working on it.  Thank you!

I just wonder if the `self_value = Quantity(self.value, self.unit)` lines (73 & 104) in the arithmetic operator wrappers should also have `subok=True` added?  This would allow direct arithmetic ops of parameters in the case of magnitude units.

Sorry that I can't insert my comment into the right place...

Reply:
This is rather dangerous since `MagUnit` based "quantities" only truly support subtraction (though addition is theoretically possible the unit of the result is not the same as that of the input units). The bigger issue is that multiplication and division are not supported e.g.:
```python
import astropy.units as u

(3 * u.ABmag) + (2 * u.ABmag)

astropy.units.core.UnitTypeError: Cannot multiply function quantities which are not dimensionless with anything.
```

Reply:
I'm confused by the example: that should work:
```
In [2]: (3 * u.ABmag) + (2 * u.ABmag)
Out[2]: <Magnitude 5. mag(AB2)>
```
But perhaps you meant to write a `*` instead of a `+`?

Anyway, I think the main problem is the one you note, that addition will typically *not* quite be what one perhaps naively expected: as one sees above, the unit of the magnitude is now `AB**2` - indeed, it makes little physical sense to add two `Magnitudes` - what does it mean to sum the magnitudes of a `V=20` and a `V=21` star? It really only makes sense to add a magnitude difference (which is dimensionless) to a magnitude.

Though I guess that might still be useful. However, I think this would need a bit more thought, so my sense for this PR specifically is just stick with what it does now -- which is very nice! And then perhaps open a separate issue about how best to deal with magnitudes in modelling more generally; I keep finding them very tricky (and I wrote the code!), and it would be good to start from some concrete examples of what one would like to fit.

Reply:
> But perhaps you meant to write a `*` instead of a `+`?

Yes... this is what I get for copying out of my terminal without checking.

Reply:
Note that at least within modeling, no arithmetic is ever done directly with a`Parameter` type (within any model or fitter), they are normally cast to a `Quantity` or pure value (depending on if operations support units or not).

Reply:
OK, I see the concerns.  Yes, that totally makes sense!  Thank you for pointing it out.

I definitely support the idea of opening a separate issue to discuss magnitudes (and other logarithmic units), in a more general sense than just in modeling.  This type of unit is so tricky.  I can think of at least one different way of magnitude addition that is practically useful, e.g., in calculating the total magnitude of a binary from its components:

```python
import astropy.units as u

mag_a = 3 * u.ABmag
mag_b = 4 * u.ABmag
mag_tot = (mag_a.to(u.AB) + mag_b.to(u.AB)).to(u.ABmag)
print(mag_tot)
```

Of course this is not to be discussed here.  You may close this conversation.  Thanks!

Reply:
@jianyangli - sounds good! As implemented, one cannot add magnitudes like you suggested, but I think it might be useful to have something that, effectively, does the operation on the underlying physical quantity. Anyway, for another issue!